src/server.c

```diff
@@ -680,6 +680,50 @@ int clientsCronResizeQueryBuffer(client *c) {
     return 0;
 }
 
+/* The client output buffer can be adjusted to better fit the memory requierments.
+ *
+ * the logic is:
+ * in case the last observed peak size of the buffer equals the buffer size - we double the size
+ * in case the last observed peak size of the buffer is less than half the buffer size - we shrink by half.
+ * in any case we will reset the peak value each cron run.
+ * The function always returns 0 as it never terminates the client. */
+int clientsCronResizeOutputBuffer(client *c) {
+
+    void *new_buf = NULL;
+    const size_t BUFFER_TARGET_SHRINK_SIZE = c->buf_usable_size/2;
+    const size_t BUFFER_TARGET_EXPEND_SIZE = c->buf_usable_size*2;
+
+    if(c->bufpos != 0)  return 0;
+
+    if (BUFFER_TARGET_SHRINK_SIZE >= PROTO_REPLY_MIN_BYTES &&
+        c->buf_peak < BUFFER_TARGET_SHRINK_SIZE )
+    {
+
```

Comment:
```suggestion
        c->buf_peak < BUFFER_TARGET_SHRINK_SIZE ) {
```

Reply:
@yoav-steinberg multi-line `if` statements have the `{` in a separate line (less confusing).
@ranshid you meant to reply to Yoav's followup on my [comment](https://github.com/redis/redis/pull/9822#discussion_r800579980).
i'll respond there

Reply:
what @yoav-steinberg requested seems more intuitive to me and it provides a simpler rules for other developers as for a single line if we request the '{' on the same line. But i would go with whatever is the current convention in our code I looked at this example https://github.com/redis/redis/blob/unstable/src/networking.c#L1152 which is like what @oranagra suggested at first. So I will change it back